include/storage/shared_memory.hpp

```diff
@@ -30,6 +30,12 @@ namespace osrm
 namespace storage
 {
 
+enum class Ownership
```

Comment:
I would move this to an own header because otherwise all classes have a dependency on quite a bit of boost interprocess headers that are not needed.